src/Uno.UI/UI/Xaml/Media/Imaging/SvgImageSource.cs

```diff
@@ -56,7 +56,12 @@ private void OnUriSourceChanged(DependencyPropertyChangedEventArgs e)
 		{
 			UnloadImageData();
 		}
-		InitFromUri(e.NewValue as Uri);
+
+		if (e.NewValue is Uri uri)
+		{
+			InitFromUri(uri);
```

Comment:
We should be consistent here and in:

https://github.com/unoplatform/uno/blob/aa761a554d8dcd6d4bdbc0c85f6df3d300585cc0/src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.cs#L35

But couple of points:

- Aside from null annotations (which is done in the other PR), we should define whether `InitFromUri` expects null or not.
- Currently, you moved few lines in `InitFromUri`, but this move is redundant if we will not be passing null.
- What is the impact of clearing `FilePath` and `AbsoluteUri` in `InitFromUri`? Is the effect visible?

I think we only need one of the two changes being done here.

Reply:
`InitFromUri(Uri)` should not allow `null`s as part of the parameters, but unfortunately, this is not something we can fully enforce on the Source Code until we have 100% null enable capabilities. 

"I think we only need one of the two changes being done here. Either guard InitFromUri calls against nulls (ie, pass it non-null), or let InitFromUri itself handle nulls."

For this fix, I do agree that making `InitFromUri()` handle the `null` is more than sufficient, I will remove it from here, but the other PR [null enabling ImageSource](https://github.com/unoplatform/uno/pull/12562) will force us to add this null validation because of the casting.

cc: @Youssef1313

Reply:
"Currently, you moved few lines in InitFromUri, but this move is redundant if we will not be passing null."
Until we are 100% null-enabled, nothing prevents this method from being called with a null value from another place